CONTRIBUTING.md

```diff
@@ -0,0 +1,151 @@
+Contributing to Mypy
+====================
+
+Welcome!  Mypy is a community project that aims to work for a wide
+swath of Python users and Python codebases.  If you're trying Mypy on
+your Python code, your experience and what you can contribute is
+important to the project's success.
+
+
+Getting started, building, and testing
+--------------------------------------
+
+If you haven't already, take a look at the project's README file and
+the [Mypy documentation](http://mypy.readthedocs.org/en/latest/), and
+try adding type annotations to your file and type-checking it with Mypy.
+
+
+Discussion
+----------
+
+If you've run into behavior in Mypy you don't understand, or you're
+having trouble working out a good way to apply it to your code, or
+you've found a bug or would like a feature it doesn't have, we want to
+hear from you!
+
+Our main forum for discussion is the project's [GitHub issue
+tracker](https://github.com/python/mypy/issues).  This is the right
+place to start a discussion of any of the above or most any other
+topic concerning the project.
+
+We have an IRC channel, `#python-mypy` on irc.freenode.net.  This is
+lightly used, but some Mypy core developers are almost always present;
+feel free to find us there and we're happy to chat.  Substantive
+technical discussion will be directed to the issue tracker.
+
+#### Code of Conduct
+
+Everyone participating in the Mypy community, and in particular in our
+issue tracker, pull requests, and IRC channel, is expected to treat
+other people with respect and more generally to follow the guidelines
+articulated in the [Python Community Code of
+Conduct](https://www.python.org/psf/codeofconduct/).
+
+
+Submitting Changes
+------------------
+
+Even more excellent than a good bug report is a fix for a bug, or the
+implementation of a much-needed new feature.  We'd love to have your
+contributions.
+
+We use the usual GitHub pull-request flow, which may be familiar to
```

Comment:
:speech_balloon: :notebook: :-P

(FTR because this was unclear -- I don't mean that I've read that legal-writing professor before and they persuaded me, rather that they're a source I found just now on the web that describes much the same approach as I've long taken based on unknown past influences. In another era I'd probably have pulled something like Garner off the bookshelf and cited that rather than gone looking on the web, but that is not an era that coexisted with GitHub pull-request threads as a discussion medium.)